DDD Workshop.ipynb

```diff
@@ -0,0 +1,547 @@
+    "    airport = line[:3]\n",
+    "    if airport in counts.keys():\n",
+    "        counts[airport] += 1\n",
+    "\n",
+    "data_file.close()\n",
+    "\n",
+    "print(counts)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Other similar backward-looking conclusions might be is the size of the planes is related to the destinations, is the occupancy is related to the plane sizes alone, etc. Most backward-looking conclusions can be drawn from the data directly, provided you are aware of the potential fallacies and have verified your data.\n",
+    "\n",
+    "Speaking of verifying your data, why is the occupied field for the YVR flights a decimal value instead of an integer number of seats like in the LAX and JFK flights? Well, that would be something we have to reason about. It would seem that the Canadian flight logs record percentage of the seats occupied instead of the number of seats occupied. Not fixing this would present a barrier to analyzing the data much further, and indeed should be fixed asap:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from math import floor\n",
+    "\n",
+    "data_file = open(\"flight_data.csv\", \"r\")\n",
+    "cleaned_data = open(\"cleaned_flight_data.csv\", \"w\")\n",
+    "\n",
+    "for line in data_file:\n",
+    "    airport = line[:3]\n",
+    "    if airport == \"YVR\":\n",
+    "        fields = line.split(\",\")\n",
+    "        num_seats = int(fields[1])\n",
+    "        occ_percent = float(fields[2])\n",
+    "        occ_count = floor(num_seats * occ_percent)\n",
+    "        line = \"{},{},{},{}\".format(fields[0], fields[1], occ_count, fields[3])\n",
+    "        \n",
+    "    cleaned_data.write(line)\n",
+    "    \n",
+    "data_file.close()\n",
+    "cleaned_data.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that any backward-looking conclusion is an observation more than it is analysis. It necessarily does not try to  make any claims about new data coming in or hypotheticals, those would be in the next section.\n",
+    "\n",
+    "\n",
+    "#### What makes a forward-looking conclusion?\n",
+    "\n",
+    "While it would feel natural to draw a dichotomy here, it would be inappropriate to do so. Forward-looking or specific conclusions depend highly on the backward-looking conclusions to make prescriptive or prospective claims. The difference between these two is that prescriptive advocates for a specific next action, the way we might want for making good, data-driven decisions, while prospective is claiming what is likely to happen. Understand that the difficulty with prescription is causality, that doing one thing would cause another, when prescriptive simply acknowledges a connection of some kind. We go back to XKCD to explore another application of bad statistical thinking:\n",
```

Comment:
Rewrite to: Understand that the difficulty with prescription is causality (doing one thing would cause another) while prescriptive simply acknowledges a connection of some kind.